src/reflect/value.go

```diff
@@ -1099,6 +1138,24 @@ func cvtUintFloat(v Value, t *rawType) Value {
 	return makeFloat(v.flags, float64(v.Uint()), t)
 }
 
+func cvtFloatInt(v Value, t *rawType) Value {
+	return makeInt(v.flags, uint64(int64(v.Float())), t)
+}
+
+func cvtFloatUint(v Value, t *rawType) Value {
+	return makeInt(v.flags, uint64(v.Float()), t)
+}
+
+func cvtFloat(v Value, t *rawType) Value {
+	if v.Type().Kind() == Float32 && t.Kind() == Float32 {
+		// Don't do any conversion if both types have underlying type float32.
+		// This avoids converting to float64 and back, which will
+		// convert a signaling NaN to a quiet NaN. See issue 36400.
```

Comment:
Ugh, NaN.
I'm not sure we even support signalling NaN at all. I believe most embedded C libraries just assume all NaNs are quiet (IMHO signalling NaN is just bad design).
Anyway, LGTM.